eng/QuarantinedTestRunsheetBuilder/QuarantinedTestRunsheetBuilder.targets

```
@@ -126,14 +120,23 @@
     </ReadLinesFromFile>
 
     <ItemGroup>
-      <_NoTestMatchesLines Include="@(FileLines)" Condition=" $([MSBuild]::ValueOrDefault(`%(FileLines.Identity)`, '').Contains(`No test matches the given testcase filter`)) " />
+      <Lines Include="@(FileLines)" >
+        <Sanitized>$([System.String]::Copy('%(FileLines.Identity)').Replace('`', ''))</Sanitized>
+      </Lines>
+
+      <_UnhandledExceptionLines Include="@(Lines)" Condition=" $([MSBuild]::ValueOrDefault(`%(Lines.Sanitized)`, '').Contains(`Unhandled exception`)) " />
+      <_ExceptionDiscoveringTestsLines Include="@(Lines)" Condition=" $([MSBuild]::ValueOrDefault(`%(Lines.Sanitized)`, '').Contains(`Exception discovering tests`)) " />
+      <_NoTestMatchesLines Include="@(Lines)" Condition=" $([MSBuild]::ValueOrDefault(`%(Lines.Sanitized)`, '').Contains(`No test matches the given testcase filter`)) " />
     </ItemGroup>
 
     <PropertyGroup>
+      <_FailedToEnumerateTests Condition=" '@(_UnhandledExceptionLines)' != '' or '@(_ExceptionDiscoveringTestsLines)' != '' ">true</_FailedToEnumerateTests>
+
       <_HasQuarantinedTests>true</_HasQuarantinedTests>
       <_HasQuarantinedTests Condition=" '@(_NoTestMatchesLines)' != '' ">false</_HasQuarantinedTests>
     </PropertyGroup>
 
+    <Error Text="Search failed: $(_ResultsFileToDisplay) [$(_TestEnvironment)]" Condition="'$(_FailedToEnumerateTests)' == 'true'" File="QuarantinedTestRunsheetBuilder" />
```

Can you please help me understand why this is useful? Does the `dotnet` process from earlier in the target not return a non-zero exit code? Or does it help to ignore some kinds of errors?

---

This is "fail fast" point to alert us that an issue was encountered while invoking `dotnet test <assembly> --list-tests` (on a specific OS we run the runsheet builder, of course).
I don't expect this error trigger at all (once we resolve the current issue with the multi-tfm tests on Linux) but if it triggers, we'll know that something is seriously wrong with our infra.

An example is the current issue with the multi-tfm tests on Linux where the tests built for net8.0 run fine but the tests targeting net9.0 fail with "Can't resolve .NET" error. For the runsheet builder run on Linux this has a material impact - instead of the "No matching tests found" string (which is used as a decision point to create a runsheet or not), we'd get the error message -> produce a runsheet -> spin up a build agent for the test project we knew we couldn't run.

---

Good idea, and thanks for the explanation! Merge away!